src/Command/Argument/Search/CreateArguments.php

```diff
@@ -0,0 +1,656 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2023 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Command\Argument\Search;
+
+use InvalidArgumentException;
+use Predis\Command\Argument\ArrayableArgument;
+
+class CreateArguments implements ArrayableArgument
+{
+    /**
+     * @var array
+     */
+    private $arguments = [];
+
+    /**
+     * @var string[]
+     */
+    private $supportedDataTypesEnum = [
+        'hash' => 'HASH',
+        'json' => 'JSON',
+    ];
+
+    /**
+     * @var string[]
+     */
+    private $sortingEnum = [
+        'asc' => 'ASC',
+        'desc' => 'DESC',
+    ];
+
+    /**
+     * Specify data type for given index. To index JSON you must have the RedisJSON module to be installed.
+     *
+     * @param  string $modifier
+     * @return $this
+     */
+    public function on(string $modifier = 'hash'): self
+    {
+        if (in_array(strtoupper($modifier), $this->supportedDataTypesEnum)) {
+            $this->arguments[] = 'ON';
+            $this->arguments[] = $this->supportedDataTypesEnum[strtolower($modifier)];
+
+            return $this;
+        }
+
+        $enumValues = implode(', ', array_values($this->supportedDataTypesEnum));
+        throw new InvalidArgumentException("Wrong modifier value given. Currently supports: {$enumValues}");
+    }
+
+    /**
+     * Adds one or more prefixes into index.
+     *
+     * @param  array $prefixes
+     * @return $this
+     */
+    public function prefix(array $prefixes): self
+    {
+        $this->arguments[] = 'PREFIX';
+        $this->arguments[] = count($prefixes);
+        $this->arguments = array_merge($this->arguments, $prefixes);
+
+        return $this;
+    }
+
+    /**
+     * Adds filter expression into index.
+     *
+     * @param  string $filter
+     * @return $this
+     */
+    public function filter(string $filter): self
+    {
+        $this->arguments[] = 'FILTER';
+        $this->arguments[] = $filter;
+
+        return $this;
+    }
+
+    /**
+     * Adds default language for documents within an index.
+     *
+     * @param  string $defaultLanguage
+     * @return $this
+     */
+    public function language(string $defaultLanguage = 'english'): self
```

Comment:
If someone calls the function and puts the default value into it, it's better not to call the function because it only adds actions that don't change anything and extends arguments[].
If possible, I think it should be written in the description of the function what the default value of that field is, and the user will call the function only if he wants to change this value.

Reply:
That how it was before, @chayim requested this changes. @chayim Should we keep it?